articles/actions.md

```diff
@@ -0,0 +1,338 @@
+
+### Cancel Request Service
+
+* **Direction**: Client calls Server
+* **Request**: Goal identifier, time stamp
+* **Response**: Goals that will be attempted to be canceled
+
+The purpose of this service is to request to cancel one or more goals on the action server.
+A cancellation request may cancel multiple goals.
+The result indicates which goals will be attempted to be canceled.
+Whether or not a goal is actually canceled is indicated by the status topic and the result service.
+
+The cancel request policy is the same as in ROS 1.
+
+* If the goal ID is empty and time is zero, cancel all goals
+* If the goal ID is empty and time is not zero, cancel all goals accepted at or before the time stamp
+* If the goal ID is not empty and time is not zero, cancel the goal with the given id regardless of the time it was accepted
+* If the goal ID is not empty and time is zero, cancel the goal with the given id and all goals accepted at or before the time stamp
+
+### Get Result Service
+
+* **Direction**: Client calls Server
+* **Request**: Goal ID
+* **Response**: Status of goal and user defined result
+
+The purpose of this service is to get the final result of a service.
+After a goal has been accepted the client should call this service to receive the result.
+The result will indicate the final status of the goal and any user defined data.
+
+Once the server sends the result to the client it should free up any resources used by the action.
+If the client never asks for the result then the server should discard the result after a timeout period.
+
+### Goal Status Topic
+
+* **Direction**: Server publishes
+* **Content**: List of in-progress goals with: Goal ID, time accepted, and an enum indicating the status
+
+This topic is published by the server to broadcast the status of goals it has accepted.
+The purpose of the topic is for introspection; it is not used by the action client.
+Messages are published when transitions from one status to another occur.
+
+The default QoS settings for a DDS middleware should be TRANSIENT LOCAL with a history size of 1.
+This allows new subscribers to always get the latest state.
+
+The possible statuses are:
+
+* *Accepted*
+  * The goal has been accepted by the action server and may now be executing
+* *Cancelling*
+  * The action server will try to cancel the indicated goal
+* *Cancelled*
+  * The action server successfully canceled the goal
+* *Succeeded*
+  * The action server successfully reached the goal
+* *Aborted*
+  * The action server failed reached the goal
+
+### Feedback Topic
+
+* **Direction**: Server publishes
+* **Content**: Goal id, user defined feedback message
+
+This topic is published by the server to send application specific progress about the goal.
+It is up to the author of the action server to decide how often to publish the feedback.
```

Comment:
It may also be interesting to be able to specify either client or server-side, whether the feedback can be lossy.

Reply:
I think the intention was to expose the feedback topic's QoS settings, but that might be complicated if different clients ask for different QoS settings, so it would likely need to be set by the server and it would be up to the clients to use compatible QoS settings.

Reply:
Yes, I can add a section about QoS settings.